src/QuickMarcEditor/QuickMarcEditor.js

```diff
@@ -31,13 +41,17 @@ const QuickMarcEditor = ({ instance, onClose, handleSubmit, initialValues, submi
       pristine={pristine}
     />
   ), [onClose, handleSubmit, submitting, pristine]);
-
+  const spySubscription = { values: true };
   const initialRecords = initialValues?.records;
 
   useEffect(() => {
     if (initialRecords) setRecords(initialRecords);
   }, [initialRecords]);
 
+  const addRecord = useCallback(({ values }) => {
+    setRecords(values.records);
```

Comment:
one more time, it will invoke `setRecords` each records change (single field value, etc)